Remove `content-type` header check for FormData body

rangermeier opened the pull request:
Closes #204 

This PR removes the check for absence of a `content-type` header when the request body is FormData or URLSearchParams.

When there are `searchParams` the updated request will not be based on the existing Request object but from `this,_options`. Otherwise the resulting new Request would omit any attached files.

sindresorhus commented:
Can you add a test for the searchParams issue?

rangermeier commented:
My first try didn't capture the issue, i.e. it didn't fail when reverting 2be9a92 .

With the changes from 0d246c9 it's actually capturing the issue plus I fixed the build errors.

sholladay commented:
Thanks for working on this. Generally, I think your solution is pretty reasonable.

Unfortunately, I think this will introduce new bugs of its own. For example, if the `input` argument is a `Request`, its headers and other settings will be ignored. Though, that's probably preferable to the current situation.

What we _really_ want to do here is update the URL of the existing `this.request`. I didn't find a way to do it when I wrote this code, other than creating a new `Request` instance.

rangermeier commented:
Thanks for reviewing, @sholladay. I added a test case for the example you gave. I also changed the code for updating the URL with the `searchParams`, so there is no loss of configuration. 

```
this.request = new globals.Request(new globals.Request(url, this.request), this._options);
```

sholladay commented on test/browser.js:
```diff
@@ -176,3 +178,56 @@ test('throws if does not support ReadableStream', withPage, async (t, page) => {
+		formData.append('file', new window.File(['bubblegum pie'], 'my-file'));
+		return window.ky(url, {
+			method: 'post',
+			searchParams: 'foo=1',
```
The test should assert that the search params are respected

rangermeier replied:
I added an assertion for the search params and moved the assertions into the server callback.

sholladay commented on test/browser.js:
```diff
@@ -176,3 +178,56 @@ test('throws if does not support ReadableStream', withPage, async (t, page) => {
+			headers: {'content-type': 'text/css'}
+		});
+		return window.ky(request, {
+			searchParams: 'foo=1'
```
Same here - make sure test fails if search params are missing from the request to the server.

rangermeier replied:
done
